### PR TITLE
feat(binder): support set_config function

### LINKED ISSUE #11146: support `set_config`
### Describe the bug

It is required for pgAdmin tool to work.
image

### Error message/log

```text
risingwave  |   2023-07-24T05:33:30.02283914Z ERROR risingwave_frontend::session: failed to handle sql:
risingwave  | SELECT set_config('bytea_output', 'hex', false) FROM pg_settings WHERE name = 'bytea_output':
risingwave  | Bind error: failed to bind expression: set_config('bytea_output', 'hex', false)
risingwave  | 
risingwave  | Caused by:
risingwave  |   Feature is not yet implemented: unsupported function: "set_config"
risingwave  | Tracking issue: https://github.com/risingwavelabs/risingwave/issues/112
risingwave  |     at src/frontend/src/session.rs:952
risingwave  |     in pgwire::pg_protocol::process_query_msg_one_stmt with session_id: 0, stmt: SELECT set_config('bytea_output', 'hex', false) FROM pg_settings WHERE name = 'bytea_output'
risingwave  | 
risingwave  |   2023-07-24T05:33:30.022951602Z ERROR pgwire::pg_protocol: error when process message, error: QueryError: Bind error: failed to bind expression: set_config('bytea_output', 'hex', false)
risingwave  | 
risingwave  | Caused by:
risingwave  |   Feature is not yet implemented: unsupported function: "set_config"
risingwave  | Tracking issue: https://github.com/risingwavelabs/risingwave/issues/112
risingwave  |     at src/utils/pgwire/src/pg_protocol.rs:192
risingwave  |
```

### To Reproduce

_No response_

### Expected behavior

_No response_

### How did you deploy RisingWave?

_No response_

### The version of RisingWave

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by TennyZhuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Fix #11146

## Checklist

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- SQL commands, functions, and operators

### Release note

* support set_config function

https://www.postgresql.org/docs/current/functions-admin.html

</details>

### CODE REVIEW

File: src/frontend/src/session.rs [hunk 1]
```diff
@@ -458,7 +458,7 @@ pub struct SessionImpl {
     // Used for user authentication.
     user_authenticator: UserAuthenticator,
     /// Stores the value of configurations.
-    config_map: RwLock<ConfigMap>,
+    config_map: Arc<RwLock<ConfigMap>>,
```
Comment (fuyufjh) [inline src/frontend/src/session.rs:461]: `Rc` might be enough?

**Review by fuyufjh — APPROVED** [review]
LGTM for rest

File: src/frontend/src/session.rs (hunk 1 above)
Comment (TennyZhuang) [reply src/frontend/src/session.rs:461]: Failed to make rustc happy.

**Comment by fuyufjh** [comment]
Did you test it locally? I tried pgAdmin with your branch but still got the same error

The query was:

```sql
SET DateStyle=ISO; SET client_min_messages=notice; SELECT set_config('bytea_output','hex',false) FROM pg_settings WHERE name = 'bytea_output'; SET client_encoding='UNICODE';
```

image

**Comment by TennyZhuang** [comment]
> Did you test it locally? I tried pgAdmin with your branch but still got the same error
> 
> The query was:
> 
> ```sql
> SET DateStyle=ISO; SET client_min_messages=notice; SELECT set_config('bytea_output','hex',false) FROM pg_settings WHERE name = 'bytea_output'; SET client_encoding='UNICODE';
> ```
> 
> image

I've already supported set_config, but not bytea_output. I guess it worth another PR.

**Comment by TennyZhuang** [comment]
> still got the same error

another error :)

**Pull request merged**